appcompat-theme/src/main/java/dev/chrisbanes/accompanist/appcompattheme/AppCompatTheme.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.AmbientContext
+import androidx.core.content.res.use
+
+/**
+ * This function creates the components of a [androidx.compose.material.MaterialTheme], reading the
+ * values from the `Theme.AppCompat` Android theme.
+ *
+ * @param context The context to read the theme from.
+ * @param readColors whether the read the color palette from the [context]'s theme.
+ * @param readTypography whether to read the font family from [context]'s theme.
+ * @param shapes A set of shapes to be used by the components in this hierarchy.
+ */
+@Composable
+fun AppCompatTheme(
+    context: Context = AmbientContext.current,
+    readColors: Boolean = true,
+    readTypography: Boolean = true,
+    shapes: Shapes = MaterialTheme.shapes,
+    content: @Composable () -> Unit
+) {
+    val themeParams = remember(context.theme) {
+        context.createAppCompatTheme(
+            readColors = readColors,
+            readTypography = readTypography
+        )
+    }
+
+    MaterialTheme(
+        colors = themeParams.colors ?: MaterialTheme.colors,
+        typography = themeParams.typography ?: MaterialTheme.typography,
+        shapes = shapes,
+        content = content
+    )
+}
+
+/**
+ * This class contains the individual components of a [MaterialTheme]: [Colors], [Typography]
+ * and [Shapes].
+ */
+data class ThemeParameters(
+    val colors: Colors?,
+    val typography: Typography?
+)
+
+/**
+ * This function creates the components of a [androidx.compose.material.MaterialTheme], reading the
+ * values from the `Theme.AppCompat` Android theme.
+ *
+ * The individual components of the returned [ThemeParameters] may be `null`, depending on the
+ * matching 'read' parameter. For example, if you set [readColors] to `false`,
+ * [ThemeParameters.colors] will be null.
+ *
+ * @param readColors whether the read the color palette from the [context]'s theme.
+ * @param readTypography whether to read the font family from [context]'s theme.
+ *
+ * @return [ThemeParameters] instance containing the resulting [Colors] and [Typography]
+ */
+fun Context.createAppCompatTheme(
+    readColors: Boolean = true,
+    readTypography: Boolean = true
+): ThemeParameters = obtainStyledAttributes(R.styleable.AppCompatThemeAdapterTheme).use { ta ->
+    require(ta.hasValue(R.styleable.AppCompatThemeAdapterTheme_windowActionBar)) {
+        "createAppCompatTheme requires the host context's theme to extend Theme.AppCompat"
+    }
+
+    val colors = if (readColors) {
+        val isLightTheme = ta.getBoolean(R.styleable.AppCompatThemeAdapterTheme_isLightTheme, true)
+
+        val defaultColors = if (isLightTheme) lightColors() else darkColors()
+
+        /* First we'll read the Material color palette */
+        val primary = ta.getComposeColor(R.styleable.AppCompatThemeAdapterTheme_colorPrimary)
+        val primaryVariant = ta.getComposeColor(R.styleable.AppCompatThemeAdapterTheme_colorPrimaryDark)
```

Comment:
Add comments when there's a "mapping" e.g. we map `colorPrimaryDark` to `primaryVariant`?